syntax = "proto2";

package interface;

message Block {
    required bytes bitsets = 1;
}

message OnlineReply {
    repeated uint64 reply = 1;
}

message OfflineReply {
    required uint32 nbrsets = 1;
    repeated uint64 hints = 2;
}

message OnlineQuery {
    required uint32 height = 1;
    required uint32 bitvec = 2;
    repeated uint32 keys = 3;
    required uint32 shifts = 4;
}

message OfflineQuery {
    required uint32 nbrsets = 1;
    required uint32 setsize = 2;
    repeated uint32 offline_keys = 3;
    repeated uint32 shifts = 4;
}

enum QueryType {
    OFFLINE = 0;
    ONLINE = 1;
    REFRESH = 2;
}

message Query {
    required QueryType type = 1;
    required bytes msg = 2;
}
